### PR TITLE
[VecOps][ROOT-10009] Add ArgMax, ArgMin operations

### PR DESCRIPTION
**Pull request by JavierCVilla**

### CODE REVIEW

File: math/vecops/inc/ROOT/RVec.hxx [hunk 1]
```diff
@@ -715,6 +715,42 @@ double Mean(const RVec<T> &v)
...
+/// Example code, at the ROOT prompt:
+/// ~~~~{.cpp}
+/// using namespace ROOT::VecOps;
+/// RVec<float> v {1.f, 2.f, 4.f};
+/// auto v_argmax = ArgMax(v);
+/// v_argmax
+/// // (int) 2
+/// ~~~~
+template <typename T>
+long ArgMax(const RVec<T> &v)
```
Comment (JavierCVilla) [inline math/vecops/inc/ROOT/RVec.hxx]: Should the return type be `Long64_t` ?

Comment (amadio) [reply math/vecops/inc/ROOT/RVec.hxx]: It should probably be whatever `operator[]` takes as input.`

File: math/vecops/inc/ROOT/RVec.hxx [hunk 2]
```diff
@@ -747,6 +747,42 @@ T Min(const RVec<T> &v)
    return *std::min_element(v.begin(), v.end());
 }
 
+// Get the position of the greatest element of an RVec
+// In case of multiple occurrences of the maximum values,
+// the indices corresponding to the first occurrence are returned.
```
Comment (dpiparo) [inline math/vecops/inc/ROOT/RVec.hxx]:
```suggestion
// the index corresponding to the first occurrence is returned.
```

File: math/vecops/inc/ROOT/RVec.hxx [hunk 3]
```diff
@@ -747,6 +747,42 @@ T Min(const RVec<T> &v)
    return *std::min_element(v.begin(), v.end());
 }
 
+// Get the position of the greatest element of an RVec
```
Comment (dpiparo) [inline math/vecops/inc/ROOT/RVec.hxx]:
```suggestion
// Get the index of the greatest element of an RVec
```

File: math/vecops/inc/ROOT/RVec.hxx [hunk 4]
```diff
@@ -747,6 +747,42 @@ T Min(const RVec<T> &v)
...
+/// v_argmax
+/// // (int) 2
+/// ~~~~
+template <typename T>
+std::size_t ArgMax(const RVec<T> &v)
+{
+   return std::distance(v.begin(), std::max_element(v.begin(), v.end()));
+}
+
+// Get the position of the smallest element of an RVec
```
Comment (dpiparo) [inline math/vecops/inc/ROOT/RVec.hxx]:
```suggestion
// Get the index of the smallest element of an RVec
```

File: math/vecops/inc/ROOT/RVec.hxx [hunk 5]
```diff
@@ -747,6 +747,42 @@ T Min(const RVec<T> &v)
...
+// In case of multiple occurrences of the minimum values,
+// the indices corresponding to the first occurrence are returned.
```
Comment (dpiparo) [inline math/vecops/inc/ROOT/RVec.hxx]:
```suggestion
// the index corresponding to the first occurrence is returned.
```

**Pull request merged**